Customer Home: Add an ABTest and associated logic to display the customer home

pablinos opened the pull request:
This updates the logic for the displaying the customer home to be the
same as displaying the checklist sidebar menu item, as well as being
part of the AB test. By default only new users with an English locale
are eligible for the test, so that covers part of our criteria.

**I think this covers everything, but I'm not 100% if all the logic is correct yet**

#### Testing instructions

With various sites, set your `customerHomePage` test variation to `show` or `hide`. If it's hidden and the site is eligible to see the checklist sidebar item, then you should see the checklist item still in the sidebar. Otherwise, you will see nothing and `/home` should redirect to `/stats`.

With `show` as the variation, and the site should see the checklist sidebar item, then this should be replaced with 'Home', and `/checklist` should redirect to `/home`. Otherwise, again you will see nothing in the sidebar and `/home` should redirect to `/stats`.

pablinos committed:
Add an ABTest and associated logic to display the customer home

This updates the logic for the displaying the customer home to be the
same as displaying the checklist sidebar menu item, as well as being
part of the AB test. By default only new users with an English locale
are eligible for the test, so that covers part of our criteria.

eliorivero commented:
<img width="276" alt="Screen Shot 2019-08-20 at 14 01 58" src="https://user-images.githubusercontent.com/1041600/63367893-290e0f80-c353-11e9-9f92-a4ca513667f6.png">

When the Checklist is displayed, the icon should be the Checklist icon.

pablinos committed:
Ensure the checklist icon is displayed for the checklist menu item

eliorivero commented:
Nice, seems to be working fine.

eliorivero commented on client/my-sites/sidebar/sidebar.jsx:
```diff
@@ -185,14 +187,14 @@ export class MySitesSidebar extends Component {
+				link={ isCustomerHomeEnabled ? '/home' + siteSuffix : '/checklist' + siteSuffix }
 				onNavigate={ this.trackCustomerHomeClick }
-				materialIcon={ isEnabled( 'customer-home' ) ? 'home' : 'check_circle' }
+				materialIcon={ isCustomerHomeEnabled ? 'home' : 'check_circle' }
```
What if you consolidate the props in a single one that is later expanded?

```js
const sidebarItemProps = isCustomerHomeEnabled
    ? {
        label: translate( 'Home' ),
        link: '/home' + siteSuffix,
        materialIcon: 'home',
    }
    : {
        label: translate( 'Checklist' ),
        link: '/checklist' + siteSuffix,
        materialIcon: 'check_circle',
    };

    return (
        <SidebarItem
            ...sidebarItemProps
            tipTarget="menus"
```

pablinos replied:
Much better! I'll update that

eliorivero commented on client/my-sites/customer-home/controller.jsx:
```diff
@@ -21,3 +25,13 @@ export default function( context, next ) {
+
+export function maybeRedirect( context, next ) {
+	const state = context.store.getState();
+	const slug = getSelectedSiteSlug( state );
```
I think it's redundant to declare `state` since you'll only use it once.I think it's best to do:

```js
const slug = getSelectedSiteSlug( context.store.getState() );
```

pablinos committed:
Refactored Customer Home Link props based on feedback

pablinos committed:
Refactored call to `getState` based on feedback

pablinos committed:
Fix failing tests by mocking selector

pablinos committed:
Removed `canCurrentUserUseCustomerHome` from `state/sites/selectors`

This was causing the `isSiteEligibleForCustomerHome` to be included
somewhere during the tests, which in turn imported the `abtest` library.
That creates a user object using `userFactory` as it's imported, and
without it being properly mocked, many tests started to fail.

My guess is that something was importing all the selectors. We can
perhaps refactor this once the A/B test is complete.

pablinos committed:
Customer Home: Update My Sites link to /home where appropriate  (#35688)

* Update My Sites link to /home where appropriate

The 'My Sites' link should now go to the customer home page, when it's
available, rather than the stats page as it does now.

* Merged changes to the canCurrentUserUseCustomerHome selector

pablinos committed:
Update the checklist menu item to use the 'home' icon

eliorivero committed:
Add flag to staging so it's visible to proxied a12s.

supernovia commented:
Hi! Silly question: I can see it was merged to master, but don't see the Home page yet.  Do I need to do something else on my end to see it now?
